models/web/scratch/interactions_this_run/snowflake/snowplow_media_player_interactions_this_run.sql

```diff
@@ -0,0 +1,94 @@
+{{
+  config(
+    materialized='table',
+    tags=["this_run"],
+    sql_header=snowplow_utils.set_query_tag(var('snowplow__query_tag', 'snowplow_dbt'))
+  )
+}}
+
+with prep as (
+
+ select
+    e.event_id,
+    e.page_view_id,
+    e.domain_sessionid,
+    e.domain_userid,
+    e.page_referrer,
+    e.page_url,
+    e.unstruct_event_com_snowplowanalytics_snowplow_media_player_event_1:label::varchar as media_label,
+    round(contexts_com_snowplowanalytics_snowplow_media_player_1[0]:duration::int) as duration,
+    e.geo_region_name,
+    e.br_name,
+    e.dvce_type,
+    e.os_name,
+    e.os_timezone,
+    e.unstruct_event_com_snowplowanalytics_snowplow_media_player_event_1:type::varchar as event_type,
+    e.derived_tstamp as start_tstamp,
+    contexts_com_snowplowanalytics_snowplow_media_player_1[0]:currentTime::varchar as player_current_time,
+    coalesce(contexts_com_snowplowanalytics_snowplow_media_player_1[0]:playbackRate::varchar, 1) as playback_rate,
+    case when e.unstruct_event_com_snowplowanalytics_snowplow_media_player_event_1:type::varchar = 'ended' then 100 else contexts_com_snowplowanalytics_snowplow_media_player_1[0]:percentProgress::int end percent_progress,
+    contexts_com_snowplowanalytics_snowplow_media_player_1[0]:muted::varchar as is_muted,
+    contexts_com_snowplowanalytics_snowplow_media_player_1[0]:isLive::varchar as is_live,
+    contexts_com_snowplowanalytics_snowplow_media_player_1[0]:loop::varchar as loop,
+    contexts_com_snowplowanalytics_snowplow_media_player_1[0]:volume::varchar as volume,
+    {% if var("snowplow__enable_youtube") %}
```

Comment:
The issue was that snowplow util's combine column version macro was not enabling me to specify a field alias which made the code extra convoluted and hard to work with later on. In order to keep one logic, I kept the Snowflake one, added a helper macro for BigQuery for now until we can shift it to snowplow utils and with that I changed the code to fit the same logic. Did the same for Redshift so that all three are in line.